src/wwwroot/js/genpage/gentab/outputhistory.js

```diff
@@ -111,15 +115,33 @@ function buttonsForImage(fullsrc, src, metadata) {
                     if (div) {
                         div.remove();
                     }
-                    div = getRequiredElementById('current_image_batch').querySelector(`.image-block[data-src="${src}"]`);
-                    if (div) {
-                        removeImageBlockFromBatch(div);
+                    let moved = false;
+                    let batchDiv = getRequiredElementById('current_image_batch').querySelector(`.image-block[data-src="${src}"]`);
+                    if (batchDiv) {
+                        removeImageBlockFromBatch(batchDiv);
+                        let cur = document.getElementById('current_image_img');
+                        let currentSrc = cur?.dataset?.src;
+                        moved = currentSrc != null && currentSrc !== prevSrc;
                     }
                     let currentImage = document.getElementById('current_image_img');
                     if (currentImage && currentImage.dataset.src == src) {
                         forceShowWelcomeMessage();
                     }
-                    imageFullView.close();
+                    if (wasFullView) {
+                        if (moved) {
+                            let cur = document.getElementById('current_image_img');
+                            if (cur) {
+                                imageFullView.showImage(cur.dataset.src, cur.dataset.metadata, cur.dataset.batch_id);
+                                imageFullView.pasteState(expandedState);
+                            }
+                        }
+                        else {
+                            imageFullView.close();
+                        }
+                    }
+                    else {
+                        imageFullView.close();
```

Comment:
if already closed, close again?

Reply:
I was thinking about forcing a close, but wouldn't have anything to close anyway!